modules/pool/invest/PoolInvestModal.tsx

```diff
@@ -56,12 +57,14 @@ export function PoolInvestModal({
     const containerControls = useAnimation();
     const modalContainerRef = useRef<HTMLDivElement | null>(null);
     const lastModalBounds = useRef<DOMRect | null>(null);
+    const { clearInvestState } = useInvestState();
 
     function onModalClose() {
         if (investComplete) {
             setModalState('start');
             setInvestType(null);
         }
+        clearInvestState();
```

Comment:
Did you verify adding the `clearInvestState` in these positions does not cause unintended consequences?

Reply:
I added the `clearInvestState` here because otherwise you could run into an `undefined` error:

1. go to FotO II
2. click 'invest'
3. choose any invest option
4. choose FTM
5. close the modal
6. go to Empire State (or any other pool where FTM is not the first (most left) token)
7.  click 'invest'
8. BOOM!

I've now fixed the bug which causes this WITHOUT clearing the investState first. 
However I would suggest we still clear the investState when closing the modal. It just looks weird when you go to invest in another pool some amounts are prefilled from state.
I also like to start 'clean' when switching my invest option but opinions might differ here.

Reply:
The point where this doesn't make sense is if I'm investing into FOTO II, make it most of the way through, accidentally close the modal, reopen it, and my progress is cleared, making me quite sad.

Reply:
I have taken out the changes for updating the invest state. I will create another PR for this.